src/components/Application/ProgressBarNEW.tsx

```diff
@@ -31,29 +37,48 @@ const ProgressBarNEW: React.FC<ProgressBarProps> = ({ structure, requestRevalida
     Object.values(sections).findIndex(({ details }) => details.code === currentSectionCode) || 0
 
   const handleChangeToPage = (sectionCode: string, pageNumber: number) => {
-    if (!structure.info.isLinear)
+    if (!isLinear) {
       push(`/applicationNEW/${structure.info.serial}/${sectionCode}/Page${pageNumber}`)
+      return
+    }
 
     // Use validationMethod to check if can change to page (on linear application) OR
     // display current page with strict validation
-    requestRevalidation(({ firstIncompletePage, setStrictSectionPage }: MethodToCallProps) => {
+    requestRevalidation(({ firstStrictInvalidPage, setStrictSectionPage }: MethodToCallProps) => {
+      if (!firstStrictInvalidPage) {
+        setStrictSectionPage(null)
+        push(`/applicationNEW/${structure.info.serial}/${sectionCode}/Page${pageNumber}`)
+        return
+      }
       if (
-        firstIncompletePage &&
         checkPageIsAccessible({
           fullStructure: structure,
-          firstIncomplete: firstIncompletePage,
+          firstIncomplete: firstStrictInvalidPage,
           current: { sectionCode, pageNumber },
         })
       ) {
+        setStrictSectionPage(null)
         push(`/applicationNEW/${structure.info.serial}/${sectionCode}/Page${pageNumber}`)
-      } else setStrictSectionPage(firstIncompletePage)
+      } else {
+        setStrictSectionPage(firstStrictInvalidPage)
+        push(
+          `/applicationNEW/${structure.info.serial}/${firstStrictInvalidPage.sectionCode}/Page${firstStrictInvalidPage.pageNumber}`
+        )
+      }
     })
   }
 
-  const getPageList = (sectionCode: string, pages: { [pageNumber: string]: PageNEW }) => {
+  const getPageList = (
+    sectionCode: string,
+    pages: { [pageNumber: string]: PageNEW },
+    isIsStrictSection: boolean
+  ) => {
     const isActivePage = (sectionCode: string, pageNumber: number) =>
       currentSectionCode === sectionCode && Number(page) === pageNumber
 
+    const checkPageIsStrict = (pageNumber: string) =>
+      isIsStrictSection && strictSectionPage?.pageNumber === Number(pageNumber)
```

Comment:
```suggestion
      isStrictSection && strictSectionPage?.pageNumber === Number(pageNumber)
```

Reply:
I did one global replace to make sure we capture all of the places this was mistyped: c7093b4381515e1a7d57a6201543e8a867bc4324